shared/AppInsightsCore/src/JavaScriptSDK/BaseCore.ts

```diff
@@ -190,50 +191,18 @@ export class BaseCore implements IAppInsightsCore {
         }
     }
 
-    /**
-     * Adds a notification listener. The SDK calls methods on the listener when an appropriate notification is raised.
-     * The added plugins must raise notifications. If the plugins do not implement the notifications, then no methods will be
-     * called.
-     * @param {INotificationListener} listener - An INotificationListener object.
-     */
-    addNotificationListener(listener: INotificationListener): void {
-        if (this._notificationManager) {
-            this._notificationManager.addNotificationListener(listener);
-        }
-    }
-
-    /**
-     * Removes all instances of the listener.
-     * @param {INotificationListener} listener - INotificationListener to remove.
-     */
-    removeNotificationListener(listener: INotificationListener): void {
-        if (this._notificationManager) {
-            this._notificationManager.removeNotificationListener(listener);
-        }
-    }
-
     private _validateTelmetryItem(telemetryItem: ITelemetryItem) {
 
         if (CoreUtils.isNullOrUndefined(telemetryItem.name)) {
-            this._notifyInvalidEvent(telemetryItem);
             throw Error("telemetry name required");
         }
 
         if (CoreUtils.isNullOrUndefined(telemetryItem.time)) {
-            this._notifyInvalidEvent(telemetryItem);
```

Comment:
You need to notify if the event is dropped here, move this validation to AICore where notification manager is available